c/serverStatusService.c

```diff
@@ -0,0 +1,272 @@
+{
+  HttpService *httpService = makeGeneratedService("Server_Status_Service", "/server/agent/**");
+  httpService->authType = SERVICE_AUTH_NATIVE_WITH_SESSION_TOKEN;
+  httpService->serviceFunction = &serveStatus;
+  httpService->runInSubtask = TRUE;
+  httpService->doImpersonation = TRUE;
+  ServerAgentContext *context = (ServerAgentContext*)safeMalloc(sizeof(ServerAgentContext), "ServerAgentContext");
+  context->serverConfig = serverSettings;
+  context->productVersion[sizeof(context->productVersion) - 1] = '\0';
+  memcpy(context->productVersion, productVer, sizeof(context->productVersion) - 1);
+  httpService->userPointer = context;
+  registerHttpService(server, httpService);
+  return 0;
+}
+
+void respondWithServerConfig(HttpResponse *response, JsonObject* config){
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  jsonStartObject(out, "options");
+  jsonPrintObject(out, config);
+  jsonEndObject(out);
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+void respondWithServerRoutes(HttpResponse *response){
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  jsonStartArray(out, "links");
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/config");
+  jsonAddString(out, "rel", "config");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/log");
+  jsonAddString(out, "rel", "log");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/logLevels");
+  jsonAddString(out, "rel", "logLevels");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/environment");
+  jsonAddString(out, "rel", "environment");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonEndArray(out);
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+void respondWithLogLevels(HttpResponse *response, ServerAgentContext *context){
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  JsonObject *logLevels = jsonObjectGetObject(context->serverConfig, "logLevels");
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  if(logLevels){
+    jsonPrintObject(out, logLevels);
+  }
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+void respondWithServerEnvironment(HttpResponse *response, ServerAgentContext *context){
+  /*Information about parameters for smf_unc: https://www.ibm.com/support/knowledgecenter/SSLTBW_2.1.0/com.ibm.zos.v2r1.erbb700/smfp.htm#smfp*/
+  struct utsname unameRet;
+  char hostnameBuffer[256];
```

Comment:
You should try not to just use arbitrary numbers for the size of a character array. It would be better to define that somewhere, but this is a minor and stylistic concern.

Reply:
http://man7.org/linux/man-pages/man2/gethostname.2.html

> "SUSv2 guarantees that "Host names are limited to 255 bytes".  POSIX.1
>        guarantees that "Host names (not including the terminating null byte)
>        are limited to HOST_NAME_MAX bytes".  On Linux, HOST_NAME_MAX is
>        defined with the value 64, which has been the limit since Linux 1.0
>        (earlier kernels imposed a limit of 8 bytes)."

256 wasnt arbitrary but I will make it a define